test/models/financialDetails/DocumentDetailSpec.scala

```diff
@@ -58,6 +58,22 @@ class DocumentDetailSpec extends UnitSpec {
 
     }
 
+    "originalAmountIsNotZeroOrNegative" should {
+      "return false" when {
+        "original amount is zero" in {
+          fullDocumentDetailModel.copy(originalAmount = Some(0)).originalAmountIsNotZeroOrNegative shouldBe false
+        }
+        "original amount is negative" in {
+          fullDocumentDetailModel.copy(originalAmount = Some(-20)).originalAmountIsNotZeroOrNegative shouldBe false
+        }
+      }
+      "return true" when {
+        "original amount is positive" in {
+          fullDocumentDetailModel.copy(originalAmount = Some(20)).originalAmountIsNotZeroOrNegative shouldBe true
+        }
+      }
```

Comment:
we are missing the case where originalAmount is None - should it really be true here (and therefore will be included in the filtered results)?

Reply:
Functionally originalAmount should never be None and if backend is not passing that data then we should be throwing an exception, which is what we are doing in the view. Let me know if you think otherwise and happy to discuss.